### PR TITLE
[MERGE] portal,website_slides: posting review on slide channel

### PR DESCRIPTION
**Pull request by jem-odoo**
Impacted modules: portal, rating, website_rating, website_slides

This merge modifies the some aspect of posting a message for public 
document, in order to implement the rating (message+rating) on a slides 
channel.

We now support the use case of posting without having the right to write 
on a document, and force the author of the message (as it is required to 
be logged to post a message). The security has been improved: to do so, 
the document token is not leaked anymore. Only the signed token (by current 
partner id) is used.
The rating mixin has been modified to handle displaying average and count 
(per document), in an efficient way.

A new frontend message composer is born in website_rating (as it allow to 
post a rating with a comment). To do so, a little refactoring of the 
Portal Chatter has been done.

For more details, please have a look at the subcommits. They are well
explained and correctly split.

Task-1902304

### CODE REVIEW

File: addons/portal/static/src/js/portal_chatter.js [hunk 1]
```diff
@@ -12,6 +12,43 @@ var time = require('web.time');
...
 var _t = core._t;
 
+/**
+ * Widget PortalComposer
+ *
+ * Display the composer (according to access right)
+ *
+ */
+var PortalComposer = Widget.extend({
+    template: 'portal.Composer',
```
Comment (tde-banana-odoo) [inline addons/portal/static/src/js/portal_chatter.js]: If it is just a move, maybe keep the lowercase ? :)

File: addons/portal/static/src/xml/portal_chatter.xml [hunk 2]
```diff
@@ -15,7 +15,12 @@
         </div>
     </t>
 
-    <t t-name="portal.chatter_composer">
+    <!--
+        Widget PortalComposer (standalone)
+
+        required many options: token, res_model, res_id, ...
```
Comment (tde-banana-odoo) [inline addons/portal/static/src/xml/portal_chatter.xml]: This new comment should either be improved either removed :p .

File: addons/portal/static/src/xml/portal_chatter.xml [hunk 3]
```diff
@@ -96,13 +104,14 @@
         </div>
     </t>
 
-    <t t-name="portal.chatter">
+    <t t-name="portal.Chatter">
```
Comment (tde-banana-odoo) [inline addons/portal/static/src/xml/portal_chatter.xml]: Maybe keep original name ?

File: addons/website_rating/static/src/scss/website_rating.scss [hunk 4]
```diff
@@ -68,3 +68,15 @@ $o-w-rating-star-color: #FACC2E;
...
 }
+
+/* Rating Popup Composer */
+.o_rating_popup_composer {
+
+    .o_rating_clickable {
+        cursor: pointer;
+    }
+
+    .o_portal_chatter_avatar {
```
Comment (tde-banana-odoo) [inline addons/website_rating/static/src/scss/website_rating.scss]: Can't you use a mr-8 as class directly ? Not sure it exists.

File: addons/website_slides/models/slide_channel.py [hunk 5]
```diff
@@ -12,13 +12,14 @@ class Channel(models.Model):
...
     _description = 'Slide Channel'
-    _inherit = ['mail.thread', 'website.seo.metadata', 'website.published.multi.mixin']
+    _inherit = ['mail.thread', 'website.seo.metadata', 'website.published.multi.mixin', 'rating.mixin']
     _order = 'sequence, id'
     _order_by_strategy = {
         'most_viewed': 'total_views desc',
         'most_voted': 'likes desc',
         'latest': 'date_published desc',
     }
+    _mail_post_access = 'read'
```
[the remark this answers is not in the document]
Comment (jem-odoo) [reply addons/website_slides/models/slide_channel.py]: To allow people to post message (and rating) on the channel. Otherwse, you need to have a 'write' access to post a message.
What is the correct solution ? As we use a controller, it looks difficult. Or we will have to use the token mechanism.
In blog, we used _mail_post_access (https://github.com/odoo/odoo/blame/12.0/addons/website_blog/models/website_blog.py#L114)

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by jem-odoo** [comment]
@d-fence Y a Robert qui nous envoie vers la doc de github. What to do to merge this PR ? Thanks !

**Comment by jem-odoo** [comment]
@robodoo retry please. Cette fois c'est la bonne, on y croit !

**Comment by tde-banana-odoo** [comment]
@robodoo r+ as per specifications

**Comment by tde-banana-odoo** [comment]
@robodoo retry

**Comment by tde-banana-odoo** [comment]
@robodoo rebase-merge

**Pull request closed without merging**